src/base/bittorrent/session.cpp

```diff
@@ -1444,12 +1444,16 @@ void Session::configureNetworkInterfaces(lt::settings_pack &settingsPack)
         const QHostAddress addr {ip};
         if (!addr.isNull())
         {
-            const QString ip = ((addr.protocol() == QAbstractSocket::IPv6Protocol)
-                          ? ('[' + Utils::Net::canonicalIPv6Addr(addr).toString() + ']')
+            const bool isIPv6 = addr.protocol() == QAbstractSocket::IPv6Protocol;
+            const QString ip = (isIPv6
+                          ? (Utils::Net::canonicalIPv6Addr(addr).toString())
                           : addr.toString());
-            endpoints << (ip + portString);
 
-            if ((ip != "0.0.0.0") && (ip != "[::]"))
+            endpoints << (isIPv6
```

Comment:
I still prefer the suggested: `endpoints << ((isIPv6 ? ('[' + ip + ']') : ip) + portString);`
It is shorter and it compiles for me.

Reply:
But it's missing the portString if isIPv6 is true. Must be at least:
`((isIPv6 ? ('[' + ip + ']') + portString : ip) + portString);`

Reply:
@treysis, seems you're wrong. Your code adds port to IPv6 address two times.

Reply:
Where?
```
isIPv6
? ('[' + ip + ']') + portString  // IPv6 with port
: ip + portString;  // IPv4 with port
```

Reply:
>Where?

Read your statement carefully. You have `(isIPv6 ? ('[' + ip + ']') + portString : ip)` and you append `portString` to it.

Reply:
You're right. I misunderstood it. Fixed it now and CI passes.